invokeai/frontend/web/src/features/nodes/types/constants.ts

```diff
@@ -68,6 +36,26 @@ export const MODEL_TYPES: FieldType[] = [
   'IPAdapterModelField',
 ];
 
+/**
+ * TODO: Revise the field type naming scheme
+ *
+ * Unfortunately, due to inconsistent naming of types, we need to keep the below map objects/callbacks.
+ *
+ * Problems:
+ * - some types do not use the word "Field" in their name, e.g. "Scheduler"
+ * - primitive types use all-lowercase names, e.g. "integer"
+ * - collection and polymorphic types do not use the word "Field"
+ *
+ * If these inconsistencies were resolved, we could remove these mappings and use simple string
+ * parsing/manipulation to handle field types.
```

Comment:
I'm not entirely convinced that this is a desired outcome. I definitely have less context than you do, but on the surface I'd tend towards not wanting to rely on any particular type name structure.

Happy to save this discussion for when the time comes though!